extensions/amp-story/1.0/amp-story.js

```diff
@@ -1540,7 +1540,9 @@ export class AmpStory extends AMP.BaseElement {
    */
   unmute_() {
     const unmuteAllMedia = () => {
-      if (this.backgroundAudioEl_) {
+      // Never play the background audio when an ad is displayed.
```

Comment:
Seems like story-level background audio should still be muted/unmuted freely, but just paused when we enter AD_STATE?

Reply:
I tried this. The problem is then there is a race between pause() and reapplyMuting. Which leads to the question should we really be muting/unmuting background audio on every page click?

Reply:
What Caleb said, plus I like the idea of handling all the unmute/mute from only one place. That reduces the complexity as it makes things a lot easier to predict, and removes risks of race conditions.

Reply:
Interesting.  They technically have different behaviors: the background-audio will be at a different timecode after the ad if we do this, whereas pause will stay at the same one.  We give no guarantees about syncing story-level background audio to story or page progress (e.g. the user can go back and get their progress out of sync with the audio), but we do currently give a guarantee about continuity (i.e. the user will hear the background audio in order, starting from the beginning).

So, I guess the question is how big of a deal is breaking that guarantee vs. how big of a deal is the implementation to fix the race condition?

> Which leads to the question should we really be muting/unmuting background audio on every page click?

I think the answer to this is "no," especially now that we've discovered that muting/unmuting is potentially an expensive operation.  It's definitely knowable which things should be muted and which things should be unmuted, so perhaps we can do that?

> [...] I like the idea of handling all the unmute/mute from only one place. That reduces the complexity as it makes things a lot easier to predict, and removes risks of race conditions.

I think we can still achieve this?  If we modify `reapplyMuting_()` to really be more like `updateMutedState_(oldPage, newPage)`, I think this would just work?